docs/design/crd-validation/README.md

```diff
@@ -0,0 +1,117 @@
+## Background
+
+This proposal is to provide validation for OCS Operator CRs. Currently, the user Creates/Updates the CR, the reconciliation might fail when there is an error in the user input. It will be a better user experience if we validate the request right after user submits the changes and before the CR gets written into etcd.
+
+## User Senarios
+
+Listing out some user cases, where validators will be helpful,
+
+a. **Expanding the cluster / Adding more capacity**
+
+From backend, validator could check the existing resources and accordingly validate the request. In future, it could check for different storage platforms (like AWS, Azure, VMWares', BareMetal etc) and further advance the validation checks.
```

Comment:
If I'm understanding this correctly, this doesn't sound like a valid thing to validate, and would be almost impossible to guarantee across multiple platforms. The Kubernetes model of eventually consistency would be to let the request for additional devices go through and retry until it succeeds.